test/unit/math/prim/prob/neg_binomial_2_log_test.cpp

```diff
@@ -1,9 +1,42 @@
+#include <stan/math/prim/mat.hpp>
 #include <stan/math/prim/scal.hpp>
+#include <test/unit/math/prim/prob/vector_rng_test_helper.hpp>
+#include <test/unit/math/prim/prob/VectorIntRNGTestRig.hpp>
 #include <gtest/gtest.h>
 #include <boost/random/mersenne_twister.hpp>
 #include <boost/math/distributions.hpp>
+#include <limits>
+#include <vector>
 #include <string>
 
+class NegativeBinomial2LogTestRig : public VectorIntRNGTestRig {
+ public:
+  NegativeBinomial2LogTestRig()
+      : VectorIntRNGTestRig(10000, 10, {0, 1, 2, 3, 4, 5, 6},
+                            {-0.5, 0.0, 0.1, 1.7}, {-2, 0, 1, 2}, {}, {},
+                            {0.1, 1.1, 4.99}, {1, 2, 3}, {-3.0, -2.0, 0.0},
+                            {-3, -1, 0}) {}
+
+  template <typename T1, typename T2, typename T3, typename T_rng>
+  auto generate_samples(const T1& eta, const T2& phi, const T3&,
+                        T_rng& rng) const {
+    return stan::math::neg_binomial_2_log_rng(eta, phi, rng);
+  }
+
+  template <typename T1>
+  double pmf(int y, T1 eta, double phi, double) const {
+    return std::exp(stan::math::neg_binomial_2_log_lpmf(y, eta, phi));
+  }
+};
+
+TEST(ProbDistributionsNegativeBinomial2Log, errorCheck) {
+  check_dist_throws_all_types(NegativeBinomial2LogTestRig());
+}
+
+TEST(ProbDistributionsNegativeBinomial2Log, distributionCheck) {
+  check_counts_real_real(NegativeBinomial2LogTestRig());
+}
+
 TEST(ProbDistributionsNegBinomial2Log, error_check) {
```

Comment:
There's a pre-existing inaccuracy in this file (`neg_binomial_2_log_test.cpp`): at the end (past the part we can see here) there's a `log_matches_lpmf` block that has tests for `neg_binomial_2_lpmf`, but this file is meant for `neg_binomial_2_log_lpmf`. So, it'd be nice if you moved that block into `neg_binomial2_test.cpp`.

Reply:
Done. I move the `NegativeBinomial2LogTestRig` in a separate header file and include that header in both test files to avoid duplication.

Reply:
Nice!